integration-web/src/main/java/org/symphonyoss/integration/web/resource/ApplicationAuthenticationResource.java

```diff
@@ -54,40 +80,52 @@
   @Autowired
   private JwtAuthentication jwtAuthentication;
 
+  @Autowired
+  private IntegrationBridge integrationBridge;
+
   /**
    * Start the JWT authentication between the App and the SBE.
    * @param configurationId Application identifier.
-   * @param podUrl The POD url.
+   * @param body Request body.
    * @return The generated Token (Ta).
    */
-  @RequestMapping(value = "/authenticate")
-  public ResponseEntity authenticate(@PathVariable String configurationId,
-      @RequestParam String podUrl) throws RemoteApiException {
+  @PostMapping(value = "/authenticate")
+  public ResponseEntity authenticate(@PathVariable String configurationId, @RequestBody String body) {
+    JsonNode node = getJsonNode(POD_ID, AUTHENTICATE, body);
+    String podId = node.path(POD_ID).asText();
+
+    validateRequiredParameter(podId, POD_ID, AUTHENTICATE);
 
     // The requested POD URL must match with the current one that is being used
     String currentPodUrl = properties.getPodUrl();
-    if (!currentPodUrl.equals(podUrl)) {
+    if (!currentPodUrl.equals(podId)) {
       ErrorResponse response = new ErrorResponse(
           HttpStatus.UNAUTHORIZED.value(),
-          logMessage.getMessage(UNAUTHORIZED_URL, podUrl, currentPodUrl));
+          logMessage.getMessage(UNAUTHORIZED_URL, podId, currentPodUrl));
       return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(response);
     }
 
+    String component = getIntegrationComponent(configurationId);
+
     String token = jwtAuthentication.authenticate(configurationId);
-    AppToken appToken = new AppToken(configurationId, token, null);
+    AppToken appToken = new AppToken(component, token, null);
 
     return ResponseEntity.ok().body(appToken);
   }
 
   /**
    * Validate the provided JWT.
    * @param configurationId Application identifier.
-   * @param jwt Json Web Token containing the user/app authentication data.
-   * @return 200 OK if it's a valid pair or a 401 otherwise.
+   * @param body Request body.
+   * @return 200 OK if it's a valid JWT token or a 401 otherwise.
    */
-  @RequestMapping(value = "/validate")
-  public ResponseEntity validate(@PathVariable String configurationId, @RequestParam String jwt)
-      throws RemoteApiException {
+  @PostMapping(value = "/validate")
```

Comment:
Why is this a Post? I understand that it's because a complex object is being passed to the Integration Bridge, but there are solutions which still lies on GET. Just to know the motivation for the change.
I'll put this comment only in this method, but it's valid for the others as well.

Reply:
@aurojr This is a POST request because GET requests should be used to read information and those scenarios we are performing actions (validation stuff). I believe POST is a better option for those scenarios.

Reply:
:+1: